.github/workflows/build_and_test.yml

```diff
@@ -140,6 +143,14 @@ jobs:
         docker_args+=(-e ENABLE_HYPRE_DEVICE=${ENABLE_HYPRE_DEVICE:-CPU})
         docker_args+=(-e ENABLE_TRILINOS=${ENABLE_TRILINOS:-ON})
 
+
+        # pygeosx testing
+        ENABLE_PYGEOSX=${{ inputs.ENABLE_PYGEOSX }}
+        docker_args+=(-e ENABLE_PYGEOSX=${ENABLE_PYGEOSX:-OFF})
```

Comment:
I wanted to reduce the number of environment variables and favor direct CLI options.
Do you think we could have the shell script have some kind of `--pygeosx` flag that would trigger `pygeos`?
That would make a long list of options indeed, but in the end, it helps us to be more explicit about what we want.